extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java

```diff
@@ -478,9 +481,21 @@ void start(@Observes @Priority(Interceptor.Priority.PLATFORM_BEFORE) StartupEven
     void destroy(@Observes @BeforeDestroyed(ApplicationScoped.class) Object event) {
         if (scheduler != null) {
             try {
-                // Note that this method does not return until all currently executing jobs have completed
-                scheduler.shutdown(true);
-            } catch (SchedulerException e) {
+                // we can't just call scheduler.shutdown(true) because that could block indefinitely
+                CompletableFuture.supplyAsync(new Supplier<>() {
```

Comment:
This supplier will be executed in `ForkJoinPool#commonPool()`. Is that intentional? Shouldn't we use a different executor instead?

Reply:
It is intentional, because my thinking is that in this case it does really matter. We certainly could use our own executor if we think it makes sense.

Reply:
I wonder if we could use `scheduler.shutdown(false)` instead... @machi1990 wdyt?

Reply:
I don't think so, as we really want to at least try and wait for the scheduled tasks to complete - at least that is my understanding.

But let's see what @machi1990 says :)

Reply:
> I wonder if we could use scheduler.shutdown(false) instead... @machi1990 wdyt?

I am thinking maybe we could make it configurable and let the user opt-in if they want gracefully shutdown with `true` - the default or hard shutdown. The user could choose depending on their application need. And for the case of the issue in  https://github.com/quarkusio/quarkus/issues/28114 `false` could be used. 
I can't seem to find the issue, but I remember that when setting this to `false` it caused some errors - produced randomly from current running schedules.

Reply:
Yeah, I think that's reasonable. @mkouba WDYT?

Reply:
Here is another idea: We have timeout property that users can set, which be default would be 10 seconds or something. If set to `0`, we would not wait at all. WDYT?

Reply:
I am +1 on this. It adds an assurance that no user process will be left handing cause of this blocking call on the other hand I cannot think of how we'd do it without it being a "breaking change". But maybe the breaking change is fine and once communicated via the migration guide, we are good. The 10 seconds default is okay IMO.

Reply:
@mkouba what's your take on this?

Reply:
So we'll introduce a new config property with a default value `10` and `0` means `scheduler.shutdown(false)`? If so I'm +1 ;-).

Reply:
Yeah, that's the idea.